### PR TITLE
feat: Support passing extra args to poetry export

### LINKED ISSUE #500: Poetry support for groups dependencies
## Is your request related to a problem? Please describe.
For managing dependencies of our lambdas we're using poetry, however it makes it frustraiting that in order to build dependencies for different lambdas in one project you either need to specify separate pyproject.toml file for each lambda, even though they might share same packages, or do some tricks with `commands` in `source_path` that will install and zip dependecies for specific lambda, specially when in one pyproject.toml you can specify common packages and create separate groups of packages for specific lambda, then, it's enough to either call:
`poetry install --with=<group_name>` -- this will install common and specific lambda packages
or
`poetry export --with=<group_name> --directory=<pyproject dir> > requirements.txt` -- this will export common and lambda specific dependencies to `requirements.txt` 

## Describe the solution you'd like.
It would be nice to have groups parameter passed to poetry command, which would then include groups dependencies, and avoid a need of having multiple pyproject.toml files in one project, which would make it easier to maintain.
Also then, whenever new dependency needs to be added it's then enough to do it from root path of the project.

Tricky part with this solution might be that lambda deployment shouldn't be dependent on changes in pyproject.toml, as adding dependency to one lambda would cause rebuild of all lambdas.

### PR DESCRIPTION
**Pull request by morganchorlton3**
## Description
I have added support to pass additional poetry arguments into the source paths
```terraform
{
      path                 = "../pyproject.toml"
      poetry_install   = true
      additional_poetry_args= ["--with", "dev", ...]
}
```

closes #500

## Motivation and Context
I have a project with a mono-repo approach with multiple lambda functions in the project, I have some separate list of dependencies defined in my pyproject.toml. I want to install additional dependencies for certain lambda functions for example 

Lambda function 1 deps: 
...

Lambda function 2 deps: 
....

Common function deps: 
....

I could then install the common deps for all lambda functions and then install specific deps for each lambda functions.

## Breaking Changes
None that i am aware off

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s) (I can do this if it would help, what would be the best to update/ should i add a new example?) 

 I have tested these changes locally with a local project deploying a lambda with and without groups defined 

I am happy to do additional testing if required. 

- [x] I have executed `pre-commit run -a` on my pull request (see screenshot)

### CODE REVIEW

**Comment by pdecat** [comment]
How about simplifying things a bit, and just accept a `poetry_args` string that's directly added to the poetry command?

**Comment by morganchorlton3** [comment]
Hi @pdecat yeah I can make that change

**Comment by morganchorlton3** [comment]
> How about simplifying things a bit, and just accept a `poetry_args` string that's directly added to the poetry command?

Change applied :)

File: README.md [hunk 1]
```diff
@@ -459,6 +459,7 @@ source_path = [
 - `pip_requirements` - Controls whether to execute `pip install`. Set to `false` to disable this feature, `true` to run `pip install` with `requirements.txt` found in `path`. Or set to another filename which you want to use instead. When `source_path` is passed as a string containing a path (and not a list of maps), and `requirements.txt` is present, `pip install` is automatically executed.
 - `pip_tmp_dir` - Set the base directory to make the temporary directory for pip installs. Can be useful for Docker in Docker builds.
 - `poetry_install` - Controls whether to execute `poetry export` and `pip install`. Set to `false` to disable this feature, `true` to run `poetry export` with `pyproject.toml` and `poetry.lock` found in `path`. When `source_path` is passed as a string containing a path (and not a list of maps), and `pyproject.toml` with a build system `poetry` is present, `poetry export` and `pip install` are automatically executed.
+- `poetry_groups` - A list of poetry dependency groups to install can be left empty to default to default dependencies
```
Comment (pdecat) [inline README.md:462]: Needs to be adapted too.

File: package.py [hunk 2]
```diff
@@ -807,6 +807,7 @@ def commands_step(path, commands):
                     prefix = claim.get("prefix_in_zip")
                     pip_requirements = claim.get("pip_requirements")
                     poetry_install = claim.get("poetry_install")
+                    additional_poetry_arg = claim.get("additional_poetry_args", [])
```
Comment (pdecat) [inline package.py:810]: Just a nit, but I'd prefer having `poetry_` as the prefix, e.g. `poetry_additional_args`, or shorter `poetry_extra_args`.

Comment (pdecat) [reply package.py:810]: Also, let's keep the same naming everywhere to ease finding things.

File: package.py [hunk 3]
```diff
@@ -1183,6 +1197,17 @@ def copy_file_to_target(file, temp_dir):
...
+
+            poetry_export = [
+                poetry_exec,
+                "export",
+                "--format",
+                "requirements.txt",
+                "--output",
+                "requirements.txt",
+                "--with-credentials",
+            ] + poetry_extra_args
```
Comment (pdecat) [inline package.py:1209]: Sorry I did not notice this earlier, one additional remark: we may want to clarify that this only applies to the `poetry export` command with `poetry_export_extra_args`

**Review by pdecat — COMMENTED** [review]
LGTM overall! One last comment about the naming.

**Review by pdecat — APPROVED** [review]
🚀

**Comment by pdecat** [comment]
PR should be renamed, e.g. `feat: Support passing extra args to poetry export `.

**Comment by antonbabenko** [comment]
This PR is included in version 7.6.0 :tada:

**Comment by aleboarna** [comment]
thank you very much! :)

**Pull request merged**